pkg/kubectl/resource_filter.go

```diff
@@ -0,0 +1,100 @@
+package kubectl
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type filterEntry struct {
+	filterFunc reflect.Value
+}
+
+// FilterOptions implements PrintOptions
+type FilterOptions struct {
```

Comment:
Agree, we don't need ResourceFilter either.

Filter should accept a type of `type FilterFunc func(runtime.Object, PrintOptions) (bool, error)`.   Simply define a function `FilterDeletedPods` and then have `kubectl.Factory` have a method for `DefaultResourceFilters` that returns an array of `FilterFunc`